app/models/language_model.rb

```diff
@@ -47,11 +49,51 @@ def created_by_current_user?
     user == Current.user
   end
 
+  def api_service_name
+    api_service.name
+  end
```

Comment:
How about `delegate :name, to: :api_service, prefix: true` and then pull it up to the other `delegate` a few lines above